calico/etcddriver/driver.py

```diff
@@ -590,6 +619,10 @@ def _on_key_updated(self, key, value):
                deletion).
         """
         if key == READY_KEY and value != "true":
+            #PLW: Should we really be starting a resync because the ready key
+            # got nuked? Seems we should sit around waiting for the ready flag.
+            # Actually, that is what we do. Better to have a different exception
+            # to avoid confusion.
             _log.warning("Ready key no longer set to true, triggering resync.")
             raise ResyncRequired()
```

Comment:
Added a comment here to explain that a resync starts by polling the Ready flag.  Should make it clear why that's the right thing to do.